### PR TITLE
Nonlinear newton

### PR DESCRIPTION
**Pull request by ksil**
Added support for Newton iteration on a system of nonlinear equations. Some helper functions were updated, and example code was added to the README file.

### CODE REVIEW

File: src/Extras/autodifferentiation.jl [hunk 1]
```diff
@@ -16,6 +16,13 @@ function cumsum(d::DualFun)
     DualFun(cumsum(d.f),(I-Evaluation(rangespace(Q),leftendpoint))*Q)
 end
 
+# total definite integral of a function
+import Base.sum
```
Comment (dlfivefifty) [inline src/Extras/autodifferentiation.jl]: This should go in the list of imports in src/ApproxFun.jl (actually I'm pretty sure it's already included)

File: src/Extras/autodifferentiation.jl [hunk 2]
```diff
@@ -52,8 +59,9 @@ end
 first(d::DualFun) = DualFun(first(d.f),Evaluation(rangespace(d.J),leftendpoint)*d.J)
 last(d::DualFun) = DualFun(last(d.f),Evaluation(rangespace(d.J),rightendpoint)*d.J)
 
-jacobian(d::DualFun)=d.J
-
+jacobian(d::DualFun) = d.J
+jacobian(a::Number) = 0.0
```
Comment (dlfivefifty) [inline src/Extras/autodifferentiation.jl]: This should be `zero(a)` for type-stability

File: src/Extras/autodifferentiation.jl [hunk 3]
```diff
@@ -102,6 +110,52 @@ function newton(N,u0::Fun;maxiterations=15,tolerance=1E-15)
...
+#       NOTE: these initial Funs should be "chopped" as much as possible for optimal speed
+#
+# KSS and ZJS
+function newton(N, u0::Array{T,1}; maxiterations=15, tolerance=1E-15) where T <: Fun
+    u = copy(u0)
+    err = Inf
+    
+    numf = length(u0)   # number of functions
+    
+    Js = Array{Any,2}(undef,1,numf) # jacobians
```
Comment (dlfivefifty) [inline src/Extras/autodifferentiation.jl]: Need to check if `numf == 0` due to use of `@inbounds` below.

**Comment by ksil** [comment]
Just committed to this branch with your comments and some unit tests. Thanks!

**Pull request merged**